pkg/telemetry/config.go

```diff
@@ -0,0 +1,43 @@
+// Copyright © 2023 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package telemetry
+
+// Config represents configuration for OpenTelemetry tracing.
+type Config struct {
+	Enable            bool            `name:"enable" description:"Enable telemetry"`
+	ResourceConfig    ResourceConfig  `name:"resource-config" description:"Information about the running service"`
+	Exporter          string          `name:"exporter" description:"Telemetry exporter (jaeger, writer)"`
+	CollectorConfig   CollectorConfig `name:"collector-config" description:"Trace collector exporter configuration"`
+	WriterConfig      WriterConfig    `name:"writer-config" description:"Writer exporter configuration"`
+	SampleProbability float64         `name:"sample-probability" description:"Sampling probability. Fractions >= 1 will always sample. Fractions < 0 are treated as zero."`
+}
+
+// ResourceConfig represents information about the running service.
+type ResourceConfig struct {
```

Comment:
I don't think this is relevant. The service version should always come from `pkg/version.String()`, and the service name is always `the-things-stack` for now.

This happens because our binary can run multiple components at once, and as such having a specific service name for one component is not really possible always.